packages/react-components/react-tree/src/components/Tree/useSubtree.ts

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { getNativeElementProps, useMergedRefs } from '@fluentui/react-utilities';
+import { TreeProps, TreeState } from './Tree.types';
+import { useTreeContext_unstable, useTreeItemContext_unstable } from '../../contexts';
+
+/**
+ * Create the state required to render a sub-level Tree.
+ *
+ * @param props - props from this instance of Tree
+ * @param ref - reference to root HTMLElement of Tree
+ */
+export function useSubtree(props: TreeProps, ref: React.Ref<HTMLElement>): TreeState {
+  const contextAppearance = useTreeContext_unstable(ctx => ctx.appearance);
+  const contextSize = useTreeContext_unstable(ctx => ctx.size);
+  const { subtreeRef } = useTreeItemContext_unstable();
```

Comment:
since the tree item context includes `isActionsVisible`, any changes to that flag will cause the entire subtree to re-render. And it can't be memoized because the re-render doesn't happen as a result of a prop